flytekit/deck/renderer.py

```diff
@@ -86,3 +86,66 @@ def to_html(self, source_code: str) -> str:
         css = formatter.get_style_defs(".highlight").replace("#fff0f0", "#ffffff")
         html = highlight(source_code, PythonLexer(), formatter)
         return f"<style>{css}</style>{html}"
+
+
+class PythonDependencyRenderer:
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, title: str = "Python Dependencies"):
+        self._title = title
+
+    def to_html(self) -> str:
+        import json
+        import subprocess
+        import sys
+
+        from flytekitplugins.deck.renderer import TableRenderer
```

Comment:
This is not correct. My [suggestion](https://github.com/flyteorg/flytekit/pull/2264#discussion_r1538289174) was to move `TableRenderer` to deck/renderer.py and then use it here.

Reply:
@jasonlai1218 , sorry for flip-flopping on this. One hard requirement that we have is that we shouldn't require pandas in flytekit core. In order to make this happen, let's revert the move of `TableRenderer` from the plugin and use the `MarkdownRenderer` (which is already in core) in  the new `PythonDependencyRenderer`. We'll have to build the markdown as the table, so something like:

```
...
installed_packages = json.loads(
                subprocess.check_output([sys.executable, "-m", "pip", "list", "--format", "json"])
            )

markdown_table = "| Name | Version |\n| --- | --- |\n"

# Append each dictionary's values to the markdown string
for entry in installed_packages:
    markdown_table += f"| {entry['name']} | {entry['version']} |\n"

...

table = MarkdownRenderer().to_html(markdown_table)
...
```

Reply:
It's OK. We want the basic core to be lightweight, but also rich in functionality, so we need to find an optimal balance. It was a great conversation, and I learned a lot. Thanks again.

Reply:
I use `<br>` instead of `\n` , because `\n` not work in html
https://github.com/flyteorg/flytekit/blob/f731757836b2b4ab83946cf52aeb98386d51cdd8/flytekit/deck/renderer.py#L115-L117

result:
![Screenshot 2024-03-29 at 12 45 25 PM](https://github.com/flyteorg/flytekit/assets/14800485/0ad6561e-d6f8-4acc-84ae-c87c8c756f8b)

and `copy button` is work

Reply:
or maybe we can show this

> adlfs==2024.2.0
> aiobotocore==2.12.1
> aiohttp==3.9.3
> aioitertools==0.11.0
> aiosignal==1.3.1
> annotated-types==0.6.0
> arrow==1.3.0
> asttokens==2.4.1
> attrs==23.2.0
> azure-core==1.30.1
> azure-datalake-store==0.0.53
> azure-identity==1.15.0
> azure-storage-blob==12.19.1
> binaryornot==0.4.4
> botocore==1.34.51

Reply:
I adjusted it again. 
Current results:
![Screenshot 2024-03-29 at 2 29 20 PM](https://github.com/flyteorg/flytekit/assets/14800485/0fcafb2c-cda7-4677-b42b-5a0a4a9296bf)